### PR TITLE
Feat/scheduled actions

### PR DESCRIPTION
**Pull request by ronaldronson**
Add SDK support for scheduled actions:
- create scheduled action on the space level
- fetch scheduled actions on the space level
- delete scheduled action on scheduled action entity level

### CODE REVIEW

File: lib/entities/scheduled-action.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  /** action was canceled by a user (terminal state) */
+  canceled = 'canceled',
+}
+
+type SchedulableEntityType = 'Entry'
+type SchedulableActionType = 'publish' | 'unpublish'
+
+export type ScheduledActionSysProps = Pick<
+  MetaSysProps,
+  'id' | 'space' | 'createdAt' | 'createdBy'
```
Comment (suevalov) [inline lib/entities/scheduled-action.ts:36]: Some of these properties are being overwritten by the type below. Do we need to use `Pick<MetaSysProps>` at all?

File: lib/entities/scheduled-action.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+> & {
+  type: 'ScheduledAction'
+  id: string
+  space: Link<'Space'>
```
Comment (suevalov) [inline lib/entities/scheduled-action.ts:40]: Usually, we don't specify a concrete literal and leave as `string`. There should be `MetaLinkProps ` type.

File: lib/entities/scheduled-action.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  scheduledFor: {
+    datetime: ISO8601Timestamp
+  }
+}
+
+export interface ScheduledActionCollection {
+  sys: {
+    type: 'Array'
+  }
+  pages: {
```
Comment (suevalov) [inline lib/entities/scheduled-action.ts:63]: `BasicCursorPaginationOptions`?

File: lib/entities/scheduled-action.ts (hunk 2 above)
Comment (ronaldronson) [reply lib/entities/scheduled-action.ts]: Thanks, replaced here and in other places with `MetaLinkProps`.

File: lib/entities/scheduled-action.ts (hunk 3 above)
Comment (ronaldronson) [reply lib/entities/scheduled-action.ts]: Indeed! Fixed.

File: lib/entities/scheduled-action.ts (hunk 1 above)
Comment (ronaldronson) [reply lib/entities/scheduled-action.ts]: Agree, removed.

**Comment by ronaldronson** [comment]
Hi @suevalov, the scheduled actions entity is space scoped on purpose, and there are no plans to move it to the environment scope. As for the flat API I want to backport those new methods, is it only the matter of merging it to the `experimental` branch or some additional afford needs to be done?

**Comment by suevalov** [comment]
@ronaldronson there's some additional work to be done, but it's rather small I would say. 

An example PR: https://github.com/contentful/contentful-management.js/pull/416

**Pull request merged**